src/qibocal/protocols/signal_experiments/calibrate_state_discrimination.py

```diff
@@ -34,6 +33,23 @@ class CalibrateStateDiscriminationParameters(Parameters):
 """Custom dtype for CalibrateStateDiscrimination."""
 
 
+def _get_lo_frequency(platform: CalibrationPlatform, qubit: QubitId) -> float:
+    """Get LO frequency given QubitId.
+
+    Currently it assumes that instruments with LOs is first one.
+    """
+    controller = next(
+        iter(
+            inst
+            for inst in platform.instruments.values()
+            if isinstance(inst, Controller)
+        )
+    )
+    probe = controller.channels[platform.qubits[qubit].probe]
```

Comment:
I think it is possible to access the channel directly from the `platform`:
```suggestion
    probe = platform.channels[platform.qubits[qubit].probe]
```
which should also lift the assumption about the instrument being the first one.

Reply:
Thanks, I was looking exactly for something like this.